pkg/scheduler/reconciliation/inmemory.go

```diff
@@ -106,6 +106,22 @@ func (r *InMemoryReconciliationRepository) RemoveReconciliation(schedulingID str
 	return nil
 }
 
+func (r *InMemoryReconciliationRepository) RemoveReconciliations(schedulingIDs []string) error {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+
+	for _, schedulingID := range schedulingIDs {
+		for _, operation := range r.operations[schedulingID] {
```

Comment:
What if there is no operations? Remember this is used for tests where we can expect unusual data patterns. I still consider delegating the job to the `RemoveReconciliation(schedulingID string)` function a better option. Unless I don't see something?

Reply:
As discussed, I have extracted the common code out to a separate re-use function.